tests/compose-tests/libbasic-test.sh

```diff
@@ -3,7 +3,21 @@ basic_test() {
 if ostree --repo=${repobuild} ls -R ${treeref} /usr/etc/passwd-; then
     assert_not_reached "Found /usr/etc/passwd- backup file in tree"
 fi
-echo "ok passwd"
+echo "ok passwd no backups"
+
+validate_passwd() {
+    f=$1
+    shift
+    ostree --repo=${repobuild} cat ${treeref} /usr/lib/$f |grep -v '^root' | sort > $f.tree
+    cat composedata/$f | while read line; do
```

Comment:
I usually do e.g.:

```bash
while read line; do
   ...
done < composedata/$f
```

Reply:
Heh I thought about that...I recall in some previous PR I brought up my "useless use of cat" award I'd given out to other people.

The thing is in this case what I don't like is it breaks the normal left to right natural flow of a pipeline, since here we're piping to a shell function.

Reply:
Another reason I avoid the `cat | while` pattern is that it creates a pipeline, which spawns separate subprocesses. The nasty consequence of that is that trying to change variables in the `while` loop will seem like it doesn't work! This has burned me so often, I just stick with the other approach now.

Totally fine with the current approach here though. I agree it does break the usual flow a bit.